tracking:
    lidar_object_tracker:
        ros__parameters:
            # laser scan pre-process
            'range_max': 25.0
            'range_min': 0.25
            'angle_max': 6.2831852
            'angle_min': 0.0
            'ma_interval': 3
            'frequency': 20
            'lidar_frame': "lower2upper"
            'odom_frame': "odom"
            'cluster_points': 15
            'cluster_angle_tolerance': 0.78
            'cluster_dist_tolerance': 1.0

            # kalman filter 
            'position_noise_variance': 0.0009 # 0.03[m]^2
            'velocity_noise_variance': 0.0049 # 0.07[m]^2
            'observation_noise_variance': 0.25 # 0.2[m]^2
            'position_error_variance': 0.09 # 1[m]^2
            'velocity_error_variance': 0.49 # 1[m]^2
            # tracking threshold
            # frames/frequency = seconds
            'frames_limit': 30 
            'velocity_limit': 5.0
            'max_track_id': 256
            min_track_id: 128
            'matching_dist': 1.0 # meter
            'velocity_sta2dyn': 1.0
            'frames_sta2dyn': 15
            'frames_dyn2sta': 15

            # visualize
            'visualize': true